plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,363 @@
+package groundwork
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gwos/tcg/sdk/clients"
+	"github.com/gwos/tcg/sdk/transit"
+	"github.com/hashicorp/go-uuid"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+const (
+	defaultMonitoringRoute = "/api/monitoring?dynamic=true"
+)
+
+// Login and logout routes from Groundwork API
+const (
+	loginURL  = "/api/auth/login"
+	logoutURL = "/api/auth/logout"
+)
+
+var (
+	sampleConfig = `
+  ## URL of your groundwork instance.
+  url = "https://groundwork.example.com"
+
+  ## Agent uuid for GroundWork API Server.
+  agent_id = ""
+
+  ## Username and password to access GroundWork API.
+  username = ""
+  password = ""
+
+  ## Default display name for the host with services(metrics).
+  # default_host = "telegraf"
+
+  ## Default service state.
+  # default_service_state = "SERVICE_OK"
+
+  ## The name of the tag that contains the hostname.
+  # resource_tag = "host"
+`
+)
+
+type Groundwork struct {
+	Server              string `toml:"url"`
+	AgentID             string `toml:"agent_id"`
+	Username            string `toml:"username"`
+	Password            string `toml:"password"`
+	DefaultHost         string `toml:"default_host"`
+	DefaultServiceState string `toml:"default_service_state"`
+	ResourceTag         string `toml:"resource_tag"`
+	authToken           string
+
+	Log telegraf.Logger `toml:"-"`
+}
+
+func (g *Groundwork) SampleConfig() string {
+	return sampleConfig
+}
+
+func (g *Groundwork) Init() error {
+	if g.Server == "" {
+		return errors.New("no 'url' provided")
+	}
+	if g.AgentID == "" {
+		return errors.New("no 'agent_id' provided")
+	}
+	if g.Username == "" {
+		return errors.New("no 'username' provided")
+	}
+	if g.Password == "" {
+		return errors.New("no 'password' provided")
+	}
+	if g.DefaultHost == "" {
+		return errors.New("no 'default_host' provided")
+	}
+	if g.ResourceTag == "" {
+		return errors.New("no 'resource_tag' provided")
+	}
+	if !validStatus(g.DefaultServiceState) {
+		return errors.New("invalid 'default_service_state' provided")
+	}
```

Comment:
Just set the default values here.

Reply:
We made that to avoid case when user sets empty string. User's value should override value from `init()` but still can be empty string

Reply:
Right. Other plugins handle empty strings set by the user just as if they have set the default. In this case, you can move the default handling in here and just use the default if the string is empty, no matter if not set or explicitly set empty by the user.
This makes the code a bit cleaner, but if you insist on this behavior, you should at least add a comment on why you do so.

Reply:
It was like that before, but then it got changed on my request. Actually, I only requested it to be done for `DefaultServiceState`, not the others. So @VladislavSenkevich I'm fine with whatever will come out if @srebhan is fine with it as well.